### PR TITLE
Lat long conversion

### LINKED ISSUE #67: Logic to map a latitude-longitude pair to its physical address / federal region
**Describe the solution you'd like**
We need a way to convert a lat long pair into its physical address and / or federal region. We also need to be able to do the reverse: turn a physical address into its lat long pair.

List of Federal Regions and their mappings:
Region I: Connecticut, Maine, Massachusetts, New Hampshire, Rhode Island, Vermont
Region II: New Jersey, New York, Puerto Rico, US Virgin Islands
Region III: Delaware, District of Columbia, Maryland, Pennsylvania, Virginia, West Virginia
Region IV: Alabama, Florida, Georgia, Kentucky, Mississippi, North Carolina, South Carolina, Tennessee
Region V: Illinois, Indiana, Michigan, Minnesota, Ohio, Wisconsin
Region VI: Arkansas, Louisiana, New Mexico, Oklahoma, Texas
Region VII: Iowa, Kansas, Missouri, Nebraska
Region VIII: Colorado, Montana, North Dakota, South Dakota, Utah, Wyoming
Region IX: Arizona, California, Hawaii, Nevada, American Samoa, Guam, Northern Mariana Islands
Region X: Alaska, Idaho, Oregon, Washington

Helpful APIs:
https://developer.mapquest.com/documentation/geocoding-api/

**Related Issues**
#45 #47

### PR DESCRIPTION
**Pull request by rxun**
## Description

Fixes #67. Adds a new field to the Resource schema (federal region). Adds two new functions to resources.js, geocoding (address to latlong) and then reverse geocoding (latlong to address). This is used when creating a new resource, converting the address into a lat-long pair and storing it in location. It will also automatically update the appropriate federal region.

### CODE REVIEW

File: backend/routes/api/resources.js [hunk 1]
```diff
@@ -4,9 +4,42 @@ const Resource = require("../../models/Resource");
 const errorWrap = require("../../utils/error-wrap");
 const { celebrate, Joi } = require("celebrate");
 const Fuse = require("fuse.js");
+const fetch = require("node-fetch");
 const { sortByDistance, options } = require("../../utils/resource-utils");
 Joi.objectId = require("joi-objectid")(Joi);
 
+async function addressToLatLong(address) {
+  let api_latlong = `http://www.mapquestapi.com/geocoding/v1/address?key=AAAppgYd4dZiZKW2sAtBtgEUAYejofok&maxResults=5&outFormat=json&location=${address}`;
```
Comment (alan-fang) [inline backend/routes/api/resources.js]: can we move the key to `.env`

Comment (alan-fang) [reply backend/routes/api/resources.js]: and maybe also the `api_latlong` to some constants section

Comment (josh-byster) [reply backend/routes/api/resources.js]: Agreed!

File: backend/routes/api/resources.js [hunk 2]
```diff
@@ -4,9 +4,107 @@ const Resource = require("../../models/Resource");
...
+
+  const response = await fetch(api_latlong, {});
+  const responseJson = await response.json();
+
+  let lat = responseJson["results"][0]["locations"][0]["latLng"]["lat"];
+  let lng = responseJson["results"][0]["locations"][0]["latLng"]["lng"];
+
+  let state = responseJson["results"][0]["locations"][0]["adminArea3"];
+  console.log(state);
+  let region = stateToFederalRegion.find(obj => obj.State === state).Region;
```
Comment (josh-byster) [inline backend/routes/api/resources.js]: Nice use of `find` here!

File: backend/routes/api/resources.js [hunk 3]
```diff
@@ -4,9 +4,107 @@ const Resource = require("../../models/Resource");
...
+  console.log(region);
+
+  return [lat, lng, region];
```
Comment (josh-byster) [inline backend/routes/api/resources.js]: Can we return as an object instead of array, like `{latitude: lat, longitude: lng, region}`

**Pull request merged**